hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/HazelcastDoubleFunction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.operators;
+
+import com.hazelcast.sql.impl.calcite.validate.types.HazelcastInferTypes;
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+
+import static com.hazelcast.sql.impl.calcite.validate.types.HazelcastOperandTypes.notAny;
+
+/**
+ * Function that accepts a DOUBLE argument and produces a DOUBLE result.
+ */
+public class HazelcastDoubleFunction extends SqlFunction {
+    public HazelcastDoubleFunction(String name) {
+        super(
+            name,
+            SqlKind.OTHER_FUNCTION,
+            ReturnTypes.DOUBLE_NULLABLE,
+            HazelcastInferTypes.DECIMAL_IF_UNKNOWN,
```

Comment:
Kind of strange to infer it as a decimal for a function accepting only double.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/17340/commits/95442d9b91abe7324b404b3fee044505319410ce
The same problem was for RAND - we inferred DECIMAL, while the function expected BIGINT.